### PR TITLE
feat: script to fix non normalized codes

### PR DESCRIPTION
**Pull request by alexgarel**
- on file system
- in mongodb

fixes:
- #7249

### CODE REVIEW

**Comment by alexgarel** [comment]
Hum, I'm not finished because there might be some non indexed entries after that (entries that where normalized but with an integer code in mongodb…

File: scripts/fix_non_normalized_codes.pl [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+=head1 NAME
+
+fix_non_ronalized_codes - A script to fix non normalized codes
```
Comment (stephanegigandet) [inline scripts/fix_non_normalized_codes.pl:25]:
```suggestion
fix_non_normalized_codes - A script to fix non normalized codes
```

File: tests/integration/fix_non_normalized_codes.t [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		`mkdir -p $product_path`;
+		store("$product_path/$rev.sto", $product_ref);
+		symlink("$rev.sto", "$product_path/product.sto");
+	}
+	# and index in mongo
+	$products_collection->insert_one($product_ref);
+    return;
+}
+
+sub pop_fields ($product_ref, $original_ref) {
```
Comment (stephanegigandet) [inline tests/integration/fix_non_normalized_codes.t:48]: Why is it named pop_fields?

**Comment by stephanegigandet** [comment]
So if you run it on a copy of the prod mongodb database, how many products would be affected for each case?

File: tests/integration/fix_non_normalized_codes.t (hunk 2 above)
Comment (alexgarel) [reply tests/integration/fix_non_normalized_codes.t:48]: Yep, not a really good name, I'll change it.

**Review by stephanegigandet — APPROVED** [review]

**Comment by alexgarel** [comment]
Ok, I merge it, it will be easier for me to test it on staging !

**Pull request merged**